pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -201,7 +109,7 @@ func (f *ring0Factory) ClientSet() (internalclientset.Interface, error) {
 
 func (f *ring0Factory) checkMatchingServerVersion() error {
 	f.checkServerVersion.Do(func() {
-		if !f.requireMatchedServerVersion {
+		if !f.clientGetter.RequireMatchServerVersion() {
```

Comment:
yeah, don't do this.  register the flag as it was before if you must. We can keep scrubbing on the way out.